install_dev_repos.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Helper script for installing spyder and external-deps locally in editable mode.
+"""
+
+import argparse
+import os
+import sys
+from logging import Formatter, StreamHandler, getLogger
+from pathlib import Path
+from subprocess import check_output
+
+from importlib_metadata import PackageNotFoundError, distribution
+from packaging.requirements import Requirement
+
+# Remove bootstrap.py directory from sys.path, otherwise spyder's install
+# status may be incorrectly determined.
+sys.path.pop(0)
```

Comment:
```suggestion
# Remove current/script directory from sys.path[0] if added by the Python invocation,
# otherwise Spyder's install status may be incorrectly determined.
SYS_PATH_0 = Path(sys.path[0]).resolve()
if SYS_PATH_0 in (Path(__file__).resolve().parent, Path.cwd()):
    sys.path.pop(0)
```

As discussed, this will remove an arbitrary path (likely an important one) if Python, for one of several reasons, didn't add one. This fixes that.